dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/compute/strategy/GetOrCreateNetworkDomainThenCreateNodes.java

```diff
@@ -79,17 +74,21 @@ protected GetOrCreateNetworkDomainThenCreateNodes(final CreateNodeWithGroupEncod
       final DimensionDataCloudControlTemplateOptions templateOptions = template.getOptions()
             .as(DimensionDataCloudControlTemplateOptions.class);
 
-      String networkDomainName = firstNonNull(templateOptions.getNetworkDomainName(), DEFAULT_NETWORK_DOMAIN_NAME);
-      String vlanName = firstNonNull(templateOptions.getVlanName(), DEFAULT_VLAN_NAME);
+      final Set<String> networks = templateOptions.getNetworks();
+
+      String networkDomainName = firstNonNull(!networks.isEmpty() ? networks.iterator().next() : null,
+            DimensionDataCloudControlTemplateOptions.DEFAULT_NETWORK_DOMAIN_NAME);
```

Comment:
Is it possible in DD to attach a server to more than one network? (That's why the `networks` property is a list). And in that case What considerations/preconditions/limitations should be taken into account?
If not, instead of silently ignoring all but the first network, it would be better to validate that there is only one and fail otherwise.

Reply:
@nacx the change I have made appears to be incorrect. In DD model a Server can be attached to many Vlan. A Vlan is associated with a single Network Domain only. A Server can only be associated with Vlans on the **same** Network Domain.
I think that recording the lists of VLANs in  `templateOptions.network` might be a better approach. My thoughts are that we revert the change to revert the `networkDomainName` removal from `DimensionDataCloudControlTemplateOptions`
Replace `DimensionDataCloudControlTemplateOptions vlanName` with `templateOptions.network`

I do have a question though about what exactly we store in `templateOptions.network`. We could store the Vlans for a given Network Domain name `org.jclouds.dimensiondata.cloudcontrol.features.NetworkApi#listVlans(java.lang.String)` **OR** we could use a Server to retrieve this information and therefore reduce the set of Vlans we store in `templateOptions.network`. Just not sure what you think is the best approach?

Reply:
>I think that recording the lists of VLANs in templateOptions.network might be a better approach. My thoughts are that we revert the change to revert the networkDomainName removal from DimensionDataCloudControlTemplateOptions
Replace DimensionDataCloudControlTemplateOptions vlanName with templateOptions.network

Sounds good. Use the DD options to configure the domain name, and read the VLAN names from the `templateOptions.networks` option.

>OR we could use a Server to retrieve this information and therefore reduce the set of Vlans we store in templateOptions.network. Just not sure what you think is the best approach?

I'm not sure I get what you mean with this. The `templateOptions.networks` options are used to tell jclouds where to attach the servers it is about to create... Could you please clarify what do you mean?

Reply:
>Sounds good. Use the DD options to configure the domain name, and read the VLAN names from the templateOptions.networks option.

Cool @nacx this is the approach I am going to take. 

>> OR we could use a Server to retrieve this information and therefore reduce the set of Vlans we store in templateOptions.network. Just not sure what you think is the best approach?

> I'm not sure I get what you mean with this. The templateOptions.networks options are used to tell jclouds where to attach the servers it is about to create... Could you please clarify what do you mean?

I took a look a bit further at the core jclouds code and that option does not make any sense, since we are actually creating nodes here.